.github/workflows/NodeJS-dispatched.yml

```diff
@@ -0,0 +1,256 @@
+        run: npx jsdoc-to-markdown --files tools/nodejs/lib/*.js >> $GITHUB_STEP_SUMMARY
+        env:
+          npm_config_yes: true
+
+      - name: Node ${{ matrix.node }}
+        shell: bash
+        if: ${{ github.event.inputs.deploy }}
+        run: ./scripts/node_build.sh ${{ matrix.node }}
+
+  osx-nodejs:
+    name: node.js OSX
+    runs-on: macos-latest
+    needs: set-up-npm
+    strategy:
+      matrix:
+        target_arch: [ x64, arm64 ]
+        node: [ '12', '14', '16', '17', '18', '19', '20' ]
+        exclude:
+          - target_arch: arm64
+            node: 12
+          - target_arch: arm64
+            node: 14
+        # these older versions of NodeJS don't have M1 support
+
+    env:
+      TARGET_ARCH: ${{ matrix.target_arch }}
+      DUCKDB_NODE_BUILD_CACHE: 0
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+          ref: ${{ github.event.inputs.ref }} 
+
+      - name: Patch
+        shell: bash
+        run: |
+          touch .github/patches/duckdb/nodejs.patch
+          git apply .github/patches/duckdb/nodejs.patch
+
+      - name: Setup Ccache
+        uses: hendrikmuhs/ccache-action@main
+        with:
+          key: ${{ github.job }}-${{ matrix.target_arch }}
+          save: ${{ ( github.event.inputs.ref == 'refs/heads/main' || github.repository != 'duckdb/duckdb' ) && ( matrix.node == '19' ) }}
+
+      - name: Build DuckDB
+        shell: bash
+        if: matrix.target_arch == 'x64'
+        run: |
+          brew install openssl ninja
+          export OPENSSL_ROOT_DIR=`brew --prefix openssl`
+          echo $OPENSSL_ROOT_DIR
+          GEN=ninja BUILD_TPCH=1 BUILD_HTTPFS=1 STATIC_OPENSSL=1 BUILD_NODE=1 make
+
+      - name: Build DuckDB
+        shell: bash
+        if: matrix.target_arch == 'arm64'
+        run: |
+          brew install ninja
+          GEN=ninja BUILD_TPCH=1 BUILD_NODE=1 make
+
+      - name: Print platform
+        shell: bash
+        run: ./build/release/duckdb -c "PRAGMA platform;"
+
+      - name: Setup
+        shell: bash
+        run: ./scripts/node_version.sh
+        env:
+          NODE_AUTH_TOKEN: ${{secrets.NODE_AUTH_TOKEN}}
+
+      - name: Node ${{ matrix.node }}
+        shell: bash
+        if: ${{ github.event.inputs.deploy }}
+        run: ./scripts/node_build.sh ${{ matrix.node }}
+
+  win-nodejs:
+    name: node.js Windows
+    runs-on: windows-latest
+    needs: set-up-npm
+    env:
+      npm_config_msvs_version: 2019
+
+    strategy:
+      matrix:
+        node: [ '12', '14', '16', '17', '18', '19', '20' ]
+
+    steps:
+      - uses: actions/setup-python@v4
+        with:
+          python-version: '3.8'
+
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+          ref: ${{ github.event.inputs.ref }} 
+
+      - name: Patch
+        shell: bash
+        run: |
+          touch .github/patches/duckdb/nodejs.patch
+          git apply .github/patches/duckdb/nodejs.patch
```

Comment:
should this have a `--allow-empty` to allow running this without a patch?

Reply:
I am not yet sure this makes sense at all, it's mostly a one-off bandaid, likely to be deleted afterwards while finding a proper long term solution to this.

Reply:
yea fair enough, 🩹 that 💩 up!